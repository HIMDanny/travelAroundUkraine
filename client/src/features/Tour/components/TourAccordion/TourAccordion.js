import React from 'react';
import { styled } from '@mui/material/styles';
import MuiAccordion from '@mui/material/Accordion';
import MuiAccordionSummary from '@mui/material/AccordionSummary';
import MuiAccordionDetails from '@mui/material/AccordionDetails';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import { Typography } from '@mui/material';
import PropTypes from 'prop-types';

const AccordionSummary = styled(MuiAccordionSummary)(({ theme }) => ({
  position: 'relative',
  padding: 0,

  '&:before': {
    content: '""',
    position: 'absolute',
    top: 0,
    left: 0,
    height: '1px',
    width: 'calc(100% - 35px)',
    borderTop: `1px solid ${theme.palette.divider}`,
  },

  '.MuiAccordionSummary-content': {
    margin: 0,
    padding: '20px 0',

    '&.Mui-expanded': {
      margin: 0,
    },
  },

  '.MuiAccordionSummary-expandIconWrapper': {
    color: `${theme.palette.divider}`,
    position: 'absolute',
    top: '-18px',
    right: 0,
  },
}));

const AccordionDetails = styled(MuiAccordionDetails)({
  padding: '0 0 30px',
});

// eslint-disable-next-line react/prop-types
const TourAccordion = ({ title, children }) => (
  <div>
    <MuiAccordion elevation={0} square defaultExpanded>
      <AccordionSummary expandIcon={<ExpandMoreIcon fontSize="large" />}>
        <Typography variant="h2">{title}</Typography>
      </AccordionSummary>
      <AccordionDetails>{children}</AccordionDetails>
    </MuiAccordion>
  </div>
);

TourAccordion.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node,
};

TourAccordion.defaultProps = {
  title: 'Accordion',
  children: <p>Accordion text</p>,
};

export default TourAccordion;
