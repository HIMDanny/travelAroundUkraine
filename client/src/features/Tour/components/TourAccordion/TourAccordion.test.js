import { render } from '@testing-library/react';
import TourAccordion from './TourAccordion';

describe('TourAccordion snapshot testing', () => {
  test('should TourAccordion match snapshot', () => {
    const { asFragment } = render(<TourAccordion />);

    expect(asFragment()).toMatchSnapshot();
  });
});
