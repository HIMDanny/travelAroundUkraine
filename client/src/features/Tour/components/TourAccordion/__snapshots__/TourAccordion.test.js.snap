// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourAccordion snapshot testing should TourAccordion match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root Mui-expanded MuiAccordion-gutters css-i5ltjc-MuiPaper-root-MuiAccordion-root"
    >
      <div
        aria-expanded="true"
        class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-tsm144-MuiButtonBase-root-MuiAccordionSummary-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
          >
            Accordion
          </h2>
        </div>
        <div
          class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiAccordion-region"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-m733ju-MuiAccordionDetails-root"
              >
                <p>
                  Accordion text
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
